File: TypeScript/palindrome_number/9.ts
function isPalindrome(x: number): boolean {
  if (x < 0) {
    return false;
  }

  const reverse = (num: number) => {
    let result = 0;

    while (num !== 0) {
      const lastDigit = num % 10;
      result = result * 10 + lastDigit;
      num = Math.floor(num / 10);
    }

    return result;
  };

  return reverse(x) === x;
}

export { isPalindrome };

File: Golang/palindrome_number/9.go
package palindromenumber

func isPalindrome(x int) bool {
	if x < 0 {
		return false
	}

	reverse := func(n int) int {
		result := 0
		for n != 0 {
			lastDigit := n % 10
			result = result*10 + lastDigit
			n /= 10
		}
		return result
	}

	return reverse(x) == x
}
